api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -566,5 +566,17 @@
 	 */
 	@Authorized
 	public void changePasswordUsingSecretAnswer(String secretAnswer, String pw) throws APIException;
-
+	
+	/*
+	 * Sets a user's activation key
+	 * @param User The user for which the activation key will be set
+	 */
+	public User setUserActivationKey(User user);
+	
+	
+	/*
+	 * Sets the time in seconds for password reset activation key to be valid 
+	 * @param validTime time for which the token is valid
+	 */
+	public void setValidTime(int validTime);
```

Comment:
Why is tis method public? This should not be a public method on the service please